### PR TITLE
[CLD-7846] Add support for authentication and deployment to non-aws clusters

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
This PR implements the "Custom" kubernetes deployment type, where you can connect to any existing cluster via a kubecfg string (in YAML format) directly, or by entering a file path to a .kube/config file. 

The bootstrapper will read all clusters in the provided file, allowing you to choose which you'd like to work with.

Once the kube config has been set, and validated, the bootstrapper works exactly as it does with an AWS flow, except it leverages the kube config to authenticate rather than the AWS API.

In implementing this, we've added unofficial support for all cloud providers, as well as local kubernetes clusters like K3s, Docker-desktop, etc.

image

image

image

image

image
(cluster list not shown for security) 

image

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-7846

### CODE REVIEW

File: providers/custom.go [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+	}
+
+	// Create a clientcmd.ClientConfig from the rest.Config
+	clientConfig := clientcmd.NewDefaultClientConfig(*config, &clientcmd.ConfigOverrides{})
+	return clientConfig, nil
+}
+
+func (p *CustomKubeProvider) KubeClient(c context.Context, clusterName string) (*model.KubeClient, error) {
+	// if p.kubeClient == nil {
+	// 1. Get the rest.Config
```
Comment (gabrieljackson) [inline providers/custom.go:191]: Is this basically a TODO?

File: providers/custom.go [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	mattermostV1AlphaClientset, err := mmclientv1alpha1.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
```
Comment (gabrieljackson) [inline providers/custom.go:205]: We probably don't need to load this, but no harm in doing so.

**Review by gabrieljackson — APPROVED** [review]
Looks good!

File: providers/custom.go (hunk 1 above)
Comment (nickmisasi) [reply providers/custom.go:191]: Forgot to remove. It worked really well when the only option was to paste an entire config yaml file, but once I added support for reading from disk it made more sense to read it from source properly each time. If this ends up being too many reads/re-instantiations/etc we can take another look at it

File: providers/custom.go (hunk 2 above)
Comment (nickmisasi) [reply providers/custom.go]: Removing for now - if at this point I've not used it we probably won't ever

File: providers/custom.go (hunk 1 above)
Comment (nickmisasi) [reply providers/custom.go]: Removed

**Pull request merged**